flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/operations/ddl/AlterTableOptionsOperation.java

```diff
@@ -22,15 +22,32 @@
 import org.apache.flink.table.catalog.ObjectIdentifier;
 import org.apache.flink.table.operations.OperationUtils;
 
+import org.apache.commons.lang3.StringUtils;
+
+import java.util.Map;
 import java.util.stream.Collectors;
 
-/** Operation to describe a ALTER TABLE .. SET .. statement. */
+/** Operation to describe a ALTER TABLE IF EXISTS .. SET .. statement. */
 public class AlterTableOptionsOperation extends AlterTableOperation {
     private final CatalogTable catalogTable;
+    private final Map<String, String> tableOptions;
```

Comment:
@Airblader , the `CatalogTable` isn't hold only by `AlterTableOptionsOperation`. Other operations like `AlterTableSchemaOperation` also hold the `CatalogTable`. 
IMO, the converter between the SQL and operation should not pass the `CatalogTable`. The `CatalogTable` is looked up when calling `TableEnvironmentImpl#executeInternal`. But the changes could be pushed another PR to unify updating.

Reply:
Yes, other operations do that, too, and probably shouldn't do it either. The design of operations needs to be improved overall, because both the operations and the converter contain way too much logic. I think it would be good to improve this for this operation with this PR, because we kind of need it. Otherwise we'd have either incorrectness or redundancy, and both would be a worse situation than before.

Reply:
@Airblader , OK, I would like to change for this.